openedx/core/djangoapps/content_tagging/rest_api/v1/views.py

```diff
@@ -130,8 +145,111 @@ def orgs(self, request, **_kwargs) -> Response:
 class ObjectTagOrgView(ObjectTagView):
     """
     View to create and retrieve ObjectTags for a provided Object ID (object_id).
-    This view extends the ObjectTagView to add Organization filters for the results.
 
     Refer to ObjectTagView docstring for usage details.
     """
     filter_backends = [ObjectTagTaxonomyOrgFilterBackend]
+
+
+class ObjectTagExportView(APIView):
+    """"
+    View to export a CSV with all children and tags for a given object_id.
+    """
+    def get(self, request: Request, **kwargs) -> StreamingHttpResponse:
+        """
+        Export a CSV with all children and tags for a given object_id.
+        """
+
+        def _content_generator(
+            tagged_content: TaggedContent, level: int = 0
+        ) -> Iterator[tuple[TaggedContent, int]]:
+            """
+            Generator that yields the tagged content and the level of the block
+            """
+            yield tagged_content, level
+            if tagged_content.children:
+                for child in tagged_content.children:
+                    yield from _content_generator(child, level + 1)
+
+        class Echo(object):
+            """
+            Class that implements just the write method of the file-like interface,
+            used for the streaming response.
```

Comment:
Can't you just use something built-in like `io.StringIO` ?

Reply:
We can, but this change allows us to use `StreamingHttpResponse`, as suggested [here](https://github.com/openedx/edx-platform/pull/34091#discussion_r1467090458)

Reply:
Hmm, ok.